nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -314,14 +314,16 @@ sub findStableDevPath {
 
 # Generate the swapDevices option from the currently activated swap
 # devices.
-my @swaps = read_file("/proc/swaps");
-shift @swaps;
+my @swaps = read_file("/proc/swaps", err_mode => 'carp');
```

Comment:
Because I just spent 10 minutes trying to find the docs on this:

> The err_mode option has three possible values: quiet, carp, or the default, croak. In quiet mode, all errors will be silent. In carp mode, all errors will be emitted as warnings. And, in croak mode, all errors will be emitted as exceptions. Take a look at Try::Tiny or Syntax::Keyword::Try to see how to catch exceptions.

From https://metacpan.org/pod/File::Slurp#read_file